vLLM Install: CMake Error at /tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/cmake/data/share/cmake-3.31/Modules/CMakeDetermineCUDACompiler.cmake:266 (message): Failed to detect a default CUDA architecture

**Describe the bug**
Following installation instructions on Ubuntu 22.04 [1]. 
Instructlab installs without issue, however pip install vllm@git+https://github.com/opendatahub-io/vllm@2024.08.01 fails on error

```
Building wheels for collected packages: vllm
  Building wheel for vllm (pyproject.toml) ... error
  error: subprocess-exited-with-error`
  
 × Building wheel for vllm (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [465 lines of output]
```


scrolling down this error is seen

```
-- The CXX compiler identification is GNU 11.4.0
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /usr/bin/c++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      -- Build type: RelWithDebInfo
      -- Target device: cuda
      -- Found Python: /home/cpaquin/Workspace/instructlab/venv/bin/python3 (found version "3.10.12") found components: Interpreter Development.Module Development.SABIModule
      -- Found python matching: /home/cpaquin/Workspace/instructlab/venv/bin/python3.
      -- Found CUDA: /usr/local/cuda (found version "12.8")
      -- The CUDA compiler identification is unknown
      CMake Error at /tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/cmake/data/share/cmake-3.31/Modules/CMakeDetermineCUDACompiler.cmake:266 (message):
        Failed to detect a default CUDA architecture.    
```

```
Complier Output
       Compiler output:

      Call Stack (most recent call first):
        /tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/torch/share/cmake/Caffe2/public/cuda.cmake:47 (enable_language)
        /tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/torch/share/cmake/Caffe2/Caffe2Config.cmake:87 (include)
        /tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/torch/share/cmake/Torch/TorchConfig.cmake:68 (find_package)
        CMakeLists.txt:67 (find_package)
      
```
      
      -- Configuring incomplete, errors occurred!
      Traceback (most recent call last):
        File "/home/cpaquin/Workspace/instructlab/venv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 389, in <module>
          main()
        File "/home/cpaquin/Workspace/instructlab/venv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 373, in main
          json_out["return_val"] = hook(**hook_input["kwargs"])
        File "/home/cpaquin/Workspace/instructlab/venv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 280, in build_wheel
          return _build_backend().build_wheel(
        File "/tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 438, in build_wheel
          return _build(['bdist_wheel', '--dist-info-dir', str(metadata_directory)])
        File "/tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 426, in _build
          return self._build_with_temp_dir(
        File "/tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 407, in _build_with_temp_dir
          self.run_setup()
        File "/tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 320, in run_setup
          exec(code, locals())
        File "<string>", line 459, in <module>
        File "/tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/setuptools/__init__.py", line 117, in setup
          return distutils.core.setup(**attrs)
        File "/tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 186, in setup
          return run_commands(dist)
        File "/tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 202, in run_commands
          dist.run_commands()
        File "/tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 983, in run_commands
          self.run_command(cmd)
        File "/tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 999, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 1002, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/setuptools/command/bdist_wheel.py", line 379, in run
          self.run_command("build")
        File "/tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 339, in run_command
          self.distribution.run_command(command)
        File "/tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 999, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 1002, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/setuptools/_distutils/command/build.py", line 136, in run
          self.run_command(cmd_name)
        File "/tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 339, in run_command
          self.distribution.run_command(command)
        File "/tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 999, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 1002, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/setuptools/command/build_ext.py", line 99, in run
          _build_ext.run(self)
        File "/tmp/pip-build-env-9m3kal5h/overlay/lib/python3.10/site-packages/setuptools/_distutils/command/build_ext.py", line 365, in run
          self.build_extensions()
        File "<string>", line 222, in build_extensions
        File "<string>", line 204, in configure
        File "/usr/lib/python3.10/subprocess.py", line 369, in check_call
          raise CalledProcessError(retcode, cmd)
      subprocess.CalledProcessError: Command '['cmake', '/tmp/pip-install-wouhqrel/vllm_7138e2abe55c4c8184e2a9f0de6f80f5', '-G', 'Ninja', '-DCMAKE_BUILD_TYPE=RelWithDebInfo', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=/tmp/pip-install-wouhqrel/vllm_7138e2abe55c4c8184e2a9f0de6f80f5/build/lib.linux-x86_64-cpython-310/vllm', '-DCMAKE_ARCHIVE_OUTPUT_DIRECTORY=build/temp.linux-x86_64-cpython-310', '-DVLLM_TARGET_DEVICE=cuda', '-DVLLM_PYTHON_EXECUTABLE=/home/cpaquin/Workspace/instructlab/venv/bin/python3', '-DVLLM_INSTALL_PUNICA_KERNELS=ON', '-DNVCC_THREADS=1', '-DCMAKE_JOB_POOL_COMPILE:STRING=compile', '-DCMAKE_JOB_POOLS:STRING=compile=8']' returned non-zero exit status 1.
      [end of output]
  
 ```
 note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for vllm
Failed to build vllm
ERROR: Failed to build installable wheels for some pyproject.toml based projects (vllm)

```

[1] https://docs.instructlab.ai/getting-started/linux_nvidia/#installing-ilab

**To Reproduce**
Steps to reproduce the behavior:
1. Follow installation instructions for ilab and vllm on https://docs.instructlab.ai/getting-started/linux_nvidia/#installing-ilab
2. Run "pip install vllm@git+https://github.com/opendatahub-io/vllm@2024.08.01"
3. Pip install fails and exits with errors as shown above

**Expected behavior**
pip install vllm@git+https://github.com/opendatahub-io/vllm@2024.08.01 runs and exits without error


**Device Info (please complete the following information):**
 - Hardware Specs: Intel i7-4790 CPU @ 3.60GHz, 32GB RAM,  NVIDIA GeForce RTX 3070 Ti
 - OS Version: Ubuntu 22.04
 - Python Version: Python 3.10.12
 - InstructLab Version: ilab, version 0.24.0

**Additional context**
CUDA available: True
CUDA version: 12.4
Torch version: 2.5.1+cu124
GPU count: 1
GPU name: NVIDIA GeForce RTX 3070 Ti

--- a/README.md
+++ b/README.md
@@ -176,7 +176,7 @@ For an overview of the full workflow, see the [workflow diagram](./docs/workflow
 
 #### Install with Nvidia CUDA
 
-- For the best CUDA experience, installing vLLM is necessary to serve Safetensors format models.
+- For the best CUDA experience, installing vLLM is necessary to serve Safetensors format models. You can install using pip install or downloading CUDA from [Nvidia](https://developer.nvidia.com/cuda-downloads)
 
    ```shell
    python<version> -m venv --upgrade-deps venv
@@ -439,7 +439,7 @@ For detailed documentation on the InstructLab LLMs and their functions, see the 
    ```shell
    ilab model download --hf-token <your-huggingface-token>
    ```
-  
+
    Hugging Face `hf_transfer` is enabled by default for faster downloads (lacks proxy support), and can be disabled with `HF_HUB_ENABLE_HF_TRANSFER=0`. If you have issues connecting to Hugging Face, refer to the [Hugging Face discussion forum](https://discuss.huggingface.co/) for more details.
 
 #### Downloading a specific model from a Hugging Face repository
@@ -533,7 +533,7 @@ For detailed documentation on the InstructLab LLMs and their functions, see the 
 
    ```shell
    source venv/bin/activate
-   ilab model chat --model ~/.cache/instructlab/models/mixtral-8x7b-instruct-v0.1.Q4_K_M.gguf 
+   ilab model chat --model ~/.cache/instructlab/models/mixtral-8x7b-instruct-v0.1.Q4_K_M.gguf
    ```
 
    > [!TIP]
@@ -584,7 +584,7 @@ Once you have enabled developer preview features, the first step is to convert y
 *Converting documents from a directory:* If you have all the source documents (e.g., PDF, Markdown) that you want to convert in a single local directory, you can convert them:
 
    ```shell
-      ilab rag convert --input-dir <path-to-source-document-directory> --output-dir <path-to-converted-document-directory> 
+      ilab rag convert --input-dir <path-to-source-document-directory> --output-dir <path-to-converted-document-directory>
    ```
 
 This capability has been tested with PDF, Markdown, and Microsoft Word format documents.  Any of the [formats supported by Docling](https://ds4sd.github.io/docling/reference/document_converter/#docling.document_converter.InputFormat) should work.
@@ -592,7 +592,7 @@ This capability has been tested with PDF, Markdown, and Microsoft Word format do
 *Converting documents from a taxonomy:* If you do not specify an input directory, the `convert` command will pull from the default --taxonomy-path value, unless specified otherwise.  For example, to use all of the documents in all of the knowledge nodes in the taxonomy in the default location you can run:
 
    ```shell
-   ilab rag convert --taxonomy-base=empty --output-dir <path-to-converted-document-directory> 
+   ilab rag convert --taxonomy-base=empty --output-dir <path-to-converted-document-directory>
    ```
 
 *Converting documents within SDG:* The SDG commands will convert documents along with its primary goal (to generate synthetic data).  If you ran any configuration of `ilab data generate`, then the knowledge documents in the taxonomy that generated synthetic data will already be converted to a structured form.  In this case, there is no need to run the `ilab rag convert` command again.  For example, the following command mentioned in the [🚀 Generate a synthetic dataset](#-generate-a-synthetic-dataset) section converts all of the documents referenced in the entire contents of the taxonomy in the default location (and outputs them to a location that is the default input location for the ingestion command below):
@@ -616,7 +616,7 @@ Once documents are converted, you can use the `ilab rag ingest` command to inges
 *Ingesting documents from ilab rag convert*: Alternatively, if you ran `ilab rag convert` to convert your documents, without SDG, then the output directory you used for `ilab rag convert` should be the input directory you specify for `ilab rag ingest`, e.g.:
 
    ```shell
-   ilab rag ingest --input-dir=<path-to-converted-document-directory> 
+   ilab rag ingest --input-dir=<path-to-converted-document-directory>
    ```
 
 #### Getting answers from RAG
@@ -778,13 +778,13 @@ There are a few models you need to download before running the InstructLab end-t
 - Download the `granite-7b-lab` model for training:
 
    ```shell
-   ilab model download --repository instructlab/granite-7b-lab 
+   ilab model download --repository instructlab/granite-7b-lab
    ```
 
 - Download the `prometheus-8x7b-v2.0` for *multi-phase training* and *benchmark evaluation*. This model is not required for `simple` or `full` training.
 
    ```shell
-   ilab model download --repository prometheus-eval/prometheus-8x7b-v2.0 --hf-token <your-huggingface-token> 
+   ilab model download --repository prometheus-eval/prometheus-8x7b-v2.0 --hf-token <your-huggingface-token>
    ```
 
 #### InstructLab model training pipelines

